packages/utils/src/lib/reports/generate-md-report.unit.test.ts

```diff
@@ -0,0 +1,389 @@
+  });
+
+  it('should include source position startLine', () => {
+    expect(
+      renderIssuesSection([
+        {
+          source: {
+            position: {
+              startLine: 4,
+            },
+          },
+        } as Issue,
+      ]),
+    ).toMatch('<td>4</td>');
+  });
+
+  it('should include source position startLine and endLine', () => {
+    expect(
+      renderIssuesSection([
+        {
+          source: {
+            position: {
+              startLine: 4,
+              endLine: 7,
+            },
+          },
+        } as Issue,
+      ]),
+    ).toMatch('<td>4-7</td>');
+  });
+});
+
+describe('renderTableSection', () => {
+  it('should render complete section', () => {
+    expect(
+      renderTableSection({
+        headings: [
+          { key: 'phase', label: 'Phase' },
+          { key: 'percentageLcp', label: '% of LCP' },
+          { key: 'timing', label: 'Timing' },
+        ],
+        items: [
+          {
+            phase: 'TTFB',
+            percentageLcp: '27%',
+            timing: '620 ms',
+          },
+          {
+            phase: 'Load Delay',
+            percentageLcp: '25%',
+            timing: '580 ms',
+          },
+          {
+            phase: 'Load Time',
+            percentageLcp: '41%',
+            timing: '940 ms',
+          },
+          {
+            phase: 'Render Delay',
+            percentageLcp: '6%',
+            timing: '140 ms',
+          },
+        ],
+      }),
+    ).toMatchSnapshot();
+  });
+});
+
+describe('reportToDetailsSection', () => {
+  it('should return details title only if no details are given', () => {
+    expect(
+      reportToDetailsSection({ score: 0, value: 125 } as AuditReport),
+    ).toBe('🟥 <b>125</b> (score: 0)');
+  });
+
+  it('should wrap "details" into HTML details element if in details are present', () => {
+    const md = reportToDetailsSection({
+      score: 0,
+      value: 0,
+      details: {
+        issues: [{}],
+      },
+    } as AuditReport);
+    expect(md).toMatch('<details>');
+    expect(md).toMatch('<summary>🟥 <b>0</b> (score: 0)</summary>');
+    expect(md).toMatch('</details>');
+  });
+
+  it('should display issue section if in "issues" are present', () => {
+    const md = reportToDetailsSection({
+      score: 0,
+      value: 0,
+      details: {
+        issues: [{}],
+      },
+    } as AuditReport);
+    expect(md).toMatch('<h4>Issues</h4>');
+    expect(md).not.toMatch('<h4>Table</h4>');
+  });
+
+  it('should display table section if in "table" is present', () => {
+    const md = reportToDetailsSection({
+      score: 0,
+      value: 0,
+      details: {
+        table: {
+          items: ['1', '2', '3'],
+        },
+        issues: [
+          {
+            message: '',
+            severity: 'info',
+            source: { file: '' },
+          },
+        ],
+      },
+    } as AuditReport);
+    expect(md).toMatch('<h4>Issues</h4>');
+    expect(md).not.toMatch('<h4>Table</h4>');
```

Comment:
🧪 The test lies: This test passes but does not verify what it describes. It should only include `table` not `issues` and include "Additional information", not "Issues".

Reply:
fixed